keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -288,6 +288,46 @@ There are two different stages when the kubelet updates its cache:
       already grant image access irregardless of this pull operation result for any next
       pull attempts.
 
+The kubelet uses the following Golang interface to interact with the cache:
+```go
+// ImagePullCredentials wraps credentials that can be used to pull an image.
+// One of the attributes must always be non-nil, and the others must be nil.
+type ImagePullCredentials struct {
+    KubernetesSecret *imagemanager.ImagePullSecret
+    NodeWideCredentials *bool
+    Anonymous *bool
+}
+
+// ImagePullManager keeps the state of images that were pulled and which are
+// currently still being pulled.
+// It should keep an internal state of images currently being pulled by the kubelet
+// in order to determine whether to destroy a "pulling" record should an image
+// pull fail.
+type ImagePullManager interface {
+    // RecordPullIntent records an intent to pull an image.
+    // It returns a number of current pulls requesting the same image.
+    RecordPullIntent(image string) (uint23, error)
+    // RecordImagePulled writes a record of an image being successfully pulled
+    // with ImagePullCredentials.
+    //
+    // If `credentials` is nil, it means the image was pulled successfully, but
+    // a reverification should be required upon next pull request. This can be
+    // useful in cases where the kubelet crashed in the middle of the pull but
+    // the image appears on the node after the kubelet restarts.
+    RecordImagePulled(imageRef string, credentials *ImagePullCredentials) error
+    // RecordImagePullFailed should be called if an image failed to pull.
+    RecordImagePullFailed(image string) error
```

Comment:
Maybe mention the action we take in this situation in the commends (which I presume is remove the intent record?)